drivers/modem/gsm_ppp.c

```diff
@@ -25,6 +25,7 @@ LOG_MODULE_REGISTER(modem_gsm, CONFIG_MODEM_LOG_LEVEL);
 #define GSM_CMD_READ_BUF       128
 #define GSM_CMD_AT_TIMEOUT     K_SECONDS(2)
 #define GSM_CMD_SETUP_TIMEOUT  K_SECONDS(6)
+#define GSM_REGISTRATION_TIMEOUT K_SECONDS(180)
```

Comment:
I was wondering if a Kconfig option wouldn't be better in this case. This way you can define a default value, let's say 20s but you can adjust based on field experimentation.

Reply:
I agree. It should be possible to tweak the timeout value using a config option instead of editing the code.